### PR TITLE
samples: Bluetooth: broadcast_audio_sink sample bugfix

### PR DESCRIPTION
**Pull request by erikrobstad**
Fixed handling of k_sem_take() return codes.
Also some minor cleanup in broadcast_audio samples.

Signed-off-by: Erik Robstad <erik.robstad@nordicsemi.no>

### CODE REVIEW

**Comment by erikrobstad** [comment]
Can this comment in broadcast_audio_sink be removed?:
/* TODO: Update K_FOREVER with a sane value, and handle error */

File: samples/bluetooth/broadcast_audio_sink/src/main.c [hunk 1]
```diff
@@ -168,6 +169,7 @@ static int init(void)
 
 static void reset(void)
 {
+	int err;
```
Comment (Thalley) [inline samples/bluetooth/broadcast_audio_sink/src/main.c:172]: Code guidelines require empty lines after variable declarations
```suggestion
	int err;
	
```

**Review by Thalley — CHANGES_REQUESTED** [review]
Minor fix required to pass CI, otherwise LGTM

File: samples/bluetooth/broadcast_audio_sink/src/main.c (hunk 1 above)
Comment (Thalley) [reply samples/bluetooth/broadcast_audio_sink/src/main.c:172]: Huh, this isn't failing CI. @jhedberg did we change that, or is the CI not running properly?

**Comment by Thalley** [comment]
> Can this comment in broadcast_audio_sink be removed?: /* TODO: Update K_FOREVER with a sane value, and handle error */

I don't think so. Ideally we'll need to add cleanup in case of timeout, e.g. if timing out on `err = k_sem_take(&sem_base_received, SEM_TIMEOUT);`, then we should remove the PA sync and start again from the beginning.

File: samples/bluetooth/broadcast_audio_sink/src/main.c (hunk 1 above)
Comment (erikrobstad) [reply samples/bluetooth/broadcast_audio_sink/src/main.c:172]: I added the empty line regardless of CI not failing

**Review by Thalley — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**